pkg/backup/backuper.go

```diff
@@ -0,0 +1,356 @@
+package backup
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/mattn/go-sqlite3"
+
+	"github.com/pkg/errors"
+)
+
+// BackupFilenamePrefix is the prefix used in every backup file.
+const BackupFilenamePrefix = "tbl_backup"
+
+// Backuper is the process that executes the backup process.
+type Backuper struct {
+	sourcePath, dir string
+	source          DB
+	backup          DB
+	config          *Config
+
+	fileCreator func(string, time.Time) (string, error)
```

Comment:
If you remember the previous implementation, it had global variables so some functions could be mocked. The problem with that is that we get race conditions when running tests in parallel. I switched to this approach.

So in tests, I will change how the backup file is created so I can test the name of the file.